docs/DeveloperGuide.md

```diff
@@ -387,57 +393,40 @@ Makes use of many methods from `UniqueList`, e.g. `add`, `setElement`, `remove`,
 
 **Aspect 2: `nric` field**
 
 To ensure that `Appointment`s are json serialisable for `Storage` in the same way as `Patient`s, all fields of the `Appointment` class have to be serialisable. To achieve this, an `nric` field is added to each `Patient` to uniquely identify patients currently stored in the system. When serialising an `Appointment`, the patient field stores the `nric` string of the patient instead, and when reading an `Appointment` from memory a lookup is performed on the existing list of patients before a valid `Appointment` object is created containing an existing Patient object.
 
-### **4.3 Appointment Manager**
-(Contributed by Shi Hui Ling & Reuben Teng)
-Scheduling, viewing, and otherwise dealing with appointments is a key feature area for Baymax. `AppointmentManager` maintains a `UniqueList` of all `Appointment`s in the app, and executes the logic of most appointment commands. 
+**Aspect 3: Marking of Appointment `status` attribute**
 
-`AppointmentManager` contains the methods necessary to operate on the `UniqueList` of `Appointment`s. These include:
- 1. Adding an appointment
- 2. Editing an appointment
- 3. Deleting an appointment
- 4. Finding a specific appointment by `Patient` and `DateTime`
- 5. Sorting the list of appointments
- 
-These methods are used by the `AppointmentCommand` classes to execute their logic.
+Automated the marking of an `Appointment` as `DONE` after the deadline has passed, and giving receptionists the ability to mark `Appointment`s as missed.
 
-#### 4.3.1 Rationale
+Option 1: Periodically check the `Datetime` of an `Appointment` that is still marked `UPCOMING` against the current `Datetime`, marking it as `DONE` if the current `Datetime` is after that of the `Appointment`.
 
-The `AppointmentManager` class contains a summary of all the "back-end" logic of appointment commands on the app's `UniqueList` of `Appointment`s. This follows the SRP, as everything related to the execution of appointment commands can be found here. This also forces the customising of code to fit exactly the purposes needed for appointment commands, even if the methods simply call a `UniqueList` method that fulfills the exact purpose.
+* Pros: Keeps stored data perpetually up-to-date.
 
-#### 4.3.2. Current Implementation
+* Cons: Constant comparision become computationally intensive when there are many `UPCOMING` appointments. 
+  
+Option 2: Check `Datetime` of `Appointment` against current `Datetime` only when `getStatus()` method called
 
-Makes use of many methods from `UniqueList`, e.g. `add`, `setElement`, `remove`, `sort`.
+* Pros: Computation is performed when the value of `status` is needed, resulting in reduced performance cost.
 
-#### 4.3.3. Design Consideration
+* Cons: `status` of `Appointment` stored within the application may be outdated, depending on when getStatus() was called.
```

Comment:
There's no more "status" stored in Appointment - status is only computed at the point of getStatus() (there is no other time status can be accessed), so whenever status of an appointment is retrieved it is always updated.

Reply:
The con I guess would be that it is a less intuitive architecture as now, appointments don't actually have an inherent status - they just have a field that specifies whether the appointment is missed, and the status is computed when necessary.

Reply:
Requested changes have been added!